Wangscape/noise/module/Bitmap.cpp

```diff
@@ -0,0 +1,116 @@
+#include "Bitmap.h"
+#include "logging/Logging.h"
+
+namespace noise
+{
+namespace module
+{
+
+namespace {
+// TODO (Serin): extract the data format from noise::RasterBase into a separate class and use it here
+static std::map<std::string, std::shared_ptr<sf::Image>> loadedImages;
+}
+
+
+Bitmap::Bitmap() : 
+    Module::Module(GetSourceModuleCount())
+{
+}
+
+const std::string & Bitmap::GetFilename() const
+{
+    return mFilename;
+}
+
+void Bitmap::SetFilename(const std::string & filename)
+{
+    // Load and process file if not already done
+    auto it = loadedImages.find(filename);
+    if (it == loadedImages.cend())
+        return;
+    else
+    {
+        auto bitmap = std::make_shared<sf::Image>();
+        // TODO get base file path from Options. How?
+        bitmap->loadFromFile(filename);
+        for (size_t y = 0; y < bitmap->getSize().y; y++)
```

Comment:
`Bitmap`'s image processing is pretty lightweight, but it's part of the solution to #32, which will use a lot of operations like this. The [draft](https://gist.github.com/serin-delaunay/a04417c5bd2ef9dad64d01f867c2ee19) uses `numpy.ndarray` with `skimage` to abstract all of these loops away, and I would hope to do something similar in the C++ implementation. I'd prefer to avoid bringing the whole of Qt into Wangscape, but a tensor/matrix/linalg library like Eigen, Armadillo, CImg, or even Boost.multi_array would probably be ideal.